docs/src/snap/howto/security/disa-stig-assessment.md

```diff
@@ -0,0 +1,6187 @@
+
+> This Finding refers to the `--insecure-port` command line argument
+> for the Kubernetes API Server service.
+> 
+> Support for the `--insecure-port` flag has been deprecated in
+> Kubernetes 1.10, and completely removed in 1.21, so this Finding
+> is Not Applicable to any versions of the k8s-snap.
+> 
+> https://github.com/kubernetes/kubernetes/issues/91506
+> 
+
+
+
+### [V-242387]: The Kubernetes Kubelet must have the "readOnlyPort" flag disabled
+
+#### Severity: High
+
+#### Status: Automated
+
+#### Upstream Finding Description:
+
+> Kubelet serves a small REST API with read access to port 10255. The read-only
+> port for Kubernetes provides no authentication or authorization security
+> control. Providing unrestricted access on port 10255 exposes Kubernetes pods
+> and containers to malicious attacks or compromise. Port 10255 is deprecated
+> and should be disabled.
+
+
+
+
+
+#### Comments:
+
+> The k8s-snap does not pass a `--config` command line argument
+> to the Kubelet service, but does explicitly pass
+> `--read-only-port=0` as a command line argument.
+> 
+> The command line arguments of Kubelet in the k8s-snap
+> are defined in the following file:
+> 
+>     /var/snap/k8s/common/args/kubelet
+> 
+
+
+#### Remediation
+
+Edit `/var/snap/k8s/common/args/kubelet` in order to set the argument
+`--read-only-port` for service `kubelet` as appropriate.
+
+Ensure it is set to: `0`
+
+Afterwards restart the `kubelet` service with:
+
+
+
+    sudo systemctl restart snap.k8s.kubelet
+
+
+
+#### Auditing (as root)
+
+Ensure that the argument `--read-only-port` for service `kubelet` is set as
+appropriate in the service's argument file `/var/snap/k8s/common/args/kubelet`.
+
+Note: this Finding allows for this argument to be UNSET as well.
+
+```bash
+grep -E -q  '\-\-read-only-port=(0)' '/var/snap/k8s/common/args/kubelet' || echo UNSET
+test $? -eq 0 && echo PASS || echo FAIL
+```
+
+In the default configuration of the `k8s-snap`, resulting output lines will start with `PASS`.
+
+The final line of the output will be `PASS`.
+
+
+
+### [V-242388]: The Kubernetes API server must have the insecure bind address not set
+
+#### Severity: High
+
+#### Status: Automated
+
+#### Upstream Finding Description:
+
+> By default, the API server will listen on two ports and addresses. One
+> address is the secure address and the other address is called the
+> "insecure bind" address and is set by default to localhost. Any requests to
+> this address bypass authentication and authorization checks. If this insecure
+> bind address is set to localhost, anyone who gains access to the host on
+> which the Control Plane is running can bypass all authorization and
+> authentication mechanisms put in place and have full control over the entire
+> cluster.
+> 
+> Close or set the insecure bind address by setting the API server's
+> "--insecure-bind-address" flag to an IP or leave it unset and ensure that the
+> "--insecure-bind-port" is not set.
+
```

Comment:
I'll try to reduce the number of irrelevant lines in the generation of the final report.